src/compiler/program.ts

```diff
@@ -1059,6 +1064,10 @@ namespace ts {
                 else {
                     // Compute the commonSourceDirectory from the input files
                     commonSourceDirectory = computeCommonSourceDirectory(files);
+                    // If we failed to find a good common directory, but outDir is specified and at least one of our files is on a windows drive/URL/other resource, add a failure
+                    if (options.outDir && commonSourceDirectory === "" && forEach(files, file => getRootLength(file.fileName) > 1)) {

```

Comment:
but apparently we cannot do this since `commonSourceDirectory` is "". Just trying to understand in what cases using `isRootedDiskPath` will yield incorrect results.